src/geometry/manifold/integral_curve.lean

```diff
@@ -0,0 +1,290 @@
+  apply has_fderiv_at.comp_has_deriv_at _ _ hd,
+  rw [hv, hv, function.comp_apply],
+  have : set.range I ∈ nhds (𝓔(I, x₀) (γ t)),
+  { rw mem_nhds_iff,
+    refine ⟨interior 𝓔(I, x₀).target, _, is_open_interior, hγ₂⟩,
+    refine set.subset.trans interior_subset _,
+    rw ext_chart_at_target,
+    exact set.inter_subset_right _ _ },
+  apply has_fderiv_within_at.has_fderiv_at _ this,
+  apply differentiable_within_at.has_fderiv_within_at,
+  apply cont_diff_within_at.differentiable_within_at _ le_top,
+  apply cont_diff_within_at_ext_coord_change,
+  apply local_equiv.mem_symm_trans_source _ hγ₁,
+  exact mem_ext_chart_source _ _
+end
+
+/-- For any continuously differentiable vector field and any chosen non-boundary point `x₀` on the
+  manifold, an integral curve `γ : ℝ → M` exists such that `γ 0 = x₀` and the tangent vector of `γ`
+  at `t` coincides with the vector field at `γ t` for all `t` within an open interval around 0.-/
+theorem exists_integral_curve_of_cont_mdiff_tangent_vector_field [proper_space E]
+  (hv : ∀ x, (v x).1 = x) (hcd : cont_mdiff I I.tangent 1 v)
+  (t₀ : ℝ) (x₀ : M) (hx : I.is_interior_point x₀) :
+  ∃ (ε > (0 : ℝ)) (γ : ℝ → M), γ t₀ = x₀ ∧ ∀ (t : ℝ), t ∈ set.Ioo (t₀ - ε) (t₀ + ε) →
+    has_mfderiv_at 𝓘(ℝ, ℝ) I γ t ((1 : ℝ →L[ℝ] ℝ).smul_right (𝓔(I.tangent, v(γ t)) (v (γ t))).2) :=
+begin
+  obtain ⟨ε, hε, γ, hf1, hf2⟩ :=
+    exists_integral_curve_of_cont_mdiff_tangent_vector_field_aux I M v hv hcd t₀ x₀ hx,
+  refine ⟨ε, hε, γ, hf1, _⟩,
+  intros t ht,
+  rw has_mfderiv_at,
+  obtain ⟨hf3, hf4, hf5, hf6⟩ := hf2 t ht,
+  use hf5,
+  rw [ext_chart_model_space_apply, written_in_ext_chart_at, ext_chart_model_space_eq_id,
+    local_equiv.refl_symm, local_equiv.refl_coe, function.comp.right_id],
+  apply has_deriv_within_at.has_fderiv_within_at,
+  apply has_deriv_at.has_deriv_within_at,
+  have hd := curve_has_deriv_at_coord_change I M v hv x₀ γ t hf3 hf4 hf6,
+  apply has_deriv_at.congr_of_eventually_eq hd,
+  rw filter.eventually_eq_iff_exists_mem,
+  refine ⟨set.Ioo (t₀ - ε) (t₀ + ε), is_open.mem_nhds is_open_Ioo ht, _⟩,
+  intros t' ht',
+  rw [function.comp_apply, function.comp_apply, function.comp_apply, local_equiv.left_inv],
+  exact (hf2 t' ht').1
+end
+
+/-- For any continuously differentiable vector field defined on a manifold without boundary and any
+  chosen starting point `x₀ : M`, an integral curve `γ : ℝ → M` exists such that `γ 0 = x₀` and the
+  tangent vector of `γ` at `t` coincides with the vector field at `γ t` for all `t` within an open
+  interval around 0. -/
+lemma exists_integral_curve_of_cont_mdiff_tangent_vector_field_of_boundaryless
+  [proper_space E] [I.boundaryless]
+  (h₁ : ∀ x, (v x).1 = x) (h₂ : cont_mdiff I I.tangent 1 v) (t₀ : ℝ) (x₀ : M) :
+  ∃ (ε > (0 : ℝ)) (γ : ℝ → M), γ t₀ = x₀ ∧ ∀ (t : ℝ), t ∈ set.Ioo (t₀ - ε) (t₀ + ε) →
+    has_mfderiv_at 𝓘(ℝ, ℝ) I γ t ((1 : ℝ →L[ℝ] ℝ).smul_right (𝓔(I.tangent, v(γ t)) (v (γ t))).2) :=
```

Comment:
The statement here is strange: the right statement should say that the derivative of gamma is `v (gamma t)` (with an `smul_right` as you do since we only have `mfderiv` and not `mderiv`), but there should be no chart in sight. My guess is that the difficulty comes from your choice for the type of `v`, as a general function from `M` to its tangent bundle (and then imposing that it's a section by putting a requirement `(v x).1 = x`). Instead, a section should probably be a function `v : Π x, tangent_bundle I x`: in this way, `v x` is by definition an element of the tangent bundle at `x`, and everything makes sense more easily.

Also, you are requiring that `v` should be `C^1` globally, but the statement should be true if `v` is just `C^1` around `x`.

Also, can you comment on the properness assumption on `E`?  Cauchy-Lipschitz works in complete spaces, so I expect this assumption should be enough, right?

There is a difficulty here that we are undergoing a refactor of vector bundles currently. By the end of this refactor, we will probably have a nice API to speak of sections, smooth sections, and so on, but it's not the case currently. So I'm not sure what to advise on this PR. If you can make it work without too much trouble with the right statement and the current API, we can probably merge it now and refactor it when the vector bundle refactor lands. @hrmacbeth , @fpvandoorn , what are your thoughts on this?

Reply:
I'll try to rewrite the type of section `v` and see if charts can be removed. I'll also relax the `C^1` condition on `v`.

The proof of `exists_is_picard_lindelof_const_of_cont_diff_on_nhds`, which states that locally `C^1` vector fields have a local solution, needs `∥v x∥` to be bounded above for `x` in a closed ball. This uses `is_compact.bdd_above_image`, which needs `proper_space E`.

I'm really looking forward to the vector bundles refactor, as the current API has been rather tricky to work with. I'm happy to use this PR to get some maths comments (which you've very helpfully given) and then rewrite it using the new API when it's ready. How can I follow the progress of its development?

Reply:
Currently, the "top" of the vector bundle refactor is #17680, but that PR depends on ~10 other open pull requests. The refactor has been moving very slowly, since both @hrmacbeth and I have been focusing more on other things recently (like mathlib4 porting).

Reply:
For `∥v x∥` to be bounded above in a small closed ball around a point `x_0`, you don't need properness: continuity at `x_0` is enough.